# Vivaria Microbiota
The scripts and data deposited in this repository represent the analysis we ran in our manuscript published in ... in 202.. Link to manuscript:

The pipelines that were followed in this study are

- QIIME (http://qiime.org)
- Phyloseq (https://joey711.github.io/phyloseq/)
- MaAsLin2 (https://huttenhower.sph.harvard.edu/maaslin/)
- Kraken2 (https://ccb.jhu.edu/software/kraken2/)
- Braken (https://ccb.jhu.edu/software/bracken/)
- InStrain (https://instrain.readthedocs.io/en/latest/)
- HAllA (https://huttenhower.sph.harvard.edu/halla/)
- metaWRAP (https://github.com/bxlab/metaWRAP)

**Data and Code Availability** All metagenomic sequencing data generated in this study has been deposited in ... with BioProject Accession  (Accession: ). The 16S rRNA amplicon sequencing dataset with processed files and the entire details of used samples can be downloaded using the following link: https://figshare.com/s/6b9e2c30c49bef0790ef (DOI: 10.6084/m9.figshare.24523324). No new code was developed in this study. We just followed the developers' suggestions in our analysis with minor optimizations.



